### ISSUE TITLE
Issue with Autolykos mining on RX570

### ISSUE THREAD

**Issue by belgeran** [Lolliedieb/lolMiner-releases#1680]
Having a weird issue with Autolykos mining
- Ubuntu 20.04
- Kernel 5.15.0.46
- AMD Drivers 22.20 (latest)
- RX570 GPU, stock firmware

lolminer works fine on etchash.. but when it comes to autolykos... not sure why but it doesn't seem to be using the memory at all, resulting in 4.75mhs only... expecting it to be around 48?
Corectrl shows that memory clock remains at 300 which is the base, and memory usage is 400+mb?

Tried with amdgpu-pro drivers from 20.40... autolykos works.. but kernel needs to be rolledback to 5.4, and corectrl doesn't work at all

**Comment by SmyDeveloper** [comment]
Hello, I am also having a similar problem with both lolminer and another miner, I switched from ETHASH to Autolykos before the merge and I have had 3 reboots of my RX 580 AMD Sapphire 8GB rig with the following error:

Unrecoverable GPU error 3.
Check the OC and UV (memory) settings on this card.
Device 3 detected as locked.
Closing the miner and trying to call an external script: ./emergency.sh (--watchdog script)

PS: With ETHASH or ETCHASH this didn't happen so often, I had miner restarts every 60-90 days, I don't know if it's something from the algorithm or directly from the miners.   

Thank's!

**Comment by Lolliedieb** [comment]
Wrt the crashes: well Autolykos and Equihash have very different profile in terms of memory and core usage. It is recommended to more or less reset oc once and start over again, because many configs that were fine on Ethash struggle with other algorithms (most need more core voltage).

The low speed on autolykos is strange ... I will check what happens if I change the driver, because on my driver (21.40) and my RX 580 8G the performance is quite normal (64.5 mh/s at 1400 core & 2100 mem)

**Comment by sameSEA** [comment]
I have same problem.
The crash always starts with "Connection lost". (though it can loose connection because of crash):

Lost connection to stratum server stratum-ergo.flypool.org:3443 or server not reachable. 
Trying to connect in 5 second(s) 
DNS over HTTPS resolve failed - switching to standard resolve 
Connected to stratum-ergo.flypool.org(172.65.222.76):3443  (TLS enabled)
TLS Handshake success 
Subscribed to stratum server
New extranonce received: 0e47 
Unrecoverable error by GPU 3.
Please check your (memory) OC & UV settings on this card.
Authorized worker: *************************************
Device 3 detected as crashed. 
Closing miner and trying to call external script: ./emergency.sh (--watchdog script) 


And this happens 2-3 times every hour

**Comment by SmyDeveloper** [comment]
> Wrt the crashes: well Autolykos and Equihash have very different profile in terms of memory and core usage. It is recommended to more or less reset oc once and start over again, because many configs that were fine on Ethash struggle with other algorithms (most need more core voltage).
> 
> The low speed on autolykos is strange ... I will check what happens if I change the driver, because on my driver (21.40) and my RX 580 8G the performance is quite normal (64.5 mh/s at 1400 core & 2100 mem)

Thanks for the answer, if I'm trying different OCs to see, I try to sacrifice hashrate to guarantee stability in the mining platform. It is seen that this problem of the RX 580 / RX 570 is happening according to what I have talked about in several groups, all of them recommended that I try to increase the VDD a bit and lower the MEM a bit.

Let's hope it's fixed.

**Comment by SmyDeveloper** [comment]
> I have same problem. The crash always starts with "Connection lost" and reconnect:
> 
> Lost connection to stratum server stratum-ergo.flypool.org:3443 or server not reachable. Trying to connect in 5 second(s) DNS over HTTPS resolve failed - switching to standard resolve Connected to stratum-ergo.flypool.org(172.65.222.76):3443 (TLS enabled) TLS Handshake success Subscribed to stratum server New extranonce received: 0e47 Unrecoverable error by GPU 3. Please check your (memory) OC & UV settings on this card. Authorized worker: ************************************* Device 3 detected as crashed. Closing miner and trying to call external script: ./emergency.sh (--watchdog script)
> 
> And this happens 2-3 times every hour. I tried with TLS or without it. I tried reducing OC far beyond known stable.

From what I have seen in some tg and discord groups, it is due to a miner problem. They all recommended me to raise the VDD a bit and lower the MEM on the GPUs that throw that error.

Try that and see how it goes.

**Comment by sameSEA** [comment]
Yes,  one hour run with significantly increased voltages shows that it could be a culprit. 
I found that other miner report losing connection as well (like srb miner).
I will report here later when run different settings (and miners)

**Comment by SmyDeveloper** [comment]
> Yes, one hour run with significantly increased voltages shows that it could be a culprit. I found that the other miner reports losing connection as well (like srb miner). I will report here later when run different settings (and miners)

It is correct, before modifying the OCs to the GPUs with problems I tried two more miners and it gave me the same error with the GPUs. Now I raised VDD a bit and lowered the MEM, they are in testing. I will do the update in 24 hours if there has been no problem and I share the OC that I am using.

**Comment by sameSEA** [comment]
Nothing worked. I gradually lowered OC and raised voltage for both core and mem from 875mV to 950mV
Core from 1200 MHz to 1150 MHz
Memory from 2100 MHz to 1750 (!!!) MHz

The frequency of crashes is same like 15-30 min, and it ALWAYS starts with "Connection lost"
When I looked this card in windows for number of memory errors - there were none at 2000 MHz

So, I think it is software error, not hardware.

**Comment by jgonzis** [comment]
> Having a weird issue with Autolykos mining
> 
> * Ubuntu 20.04
> * Kernel 5.15.0.46
> * AMD Drivers 22.20 (latest)
> * RX570 GPU, stock firmware
> 
> lolminer works fine on etchash.. but when it comes to autolykos... not sure why but it doesn't seem to be using the memory at all, resulting in 4.75mhs only... expecting it to be around 48? Corectrl shows that memory clock remains at 300 which is the base, and memory usage is 400+mb?
> 
> Tried with amdgpu-pro drivers from 20.40... autolykos works.. but kernel needs to be rolledback to 5.4, and corectrl doesn't work at all

AMD drivers is a nightmare with all the changes they do... i

**Comment by SmyDeveloper** [comment]
> > Yes, one hour run with significantly increased voltages shows that it could be a culprit. I found that the other miner reports losing connection as well (like srb miner). I will report here later when run different settings (and miners)
> 
> It is correct, before modifying the OCs to the GPUs with problems I tried two more miners and it gave me the same error with the GPUs. Now I raised VDD a bit and lowered the MEM, they are in testing. I will do the update in 24 hours if there has been no problem and I share the OC that I am using.

Hello, I'm going with update:

I keep having rig reboots, right now I am testing the following OCs for 12 AMD RX 580 SAPPHIRE PULSE 8GB Samsung Memory:

CORE 1150 VDD 850 MEM 1950 for 60.86 mh - 61.11 mh. I'll have this in evidence for 24 hours.

I've tried these OCs but no luck yet, I always get rig reboots 2 times a day:

CORE: 1250 VDD 830 MEM 2100.
CORE: 1200 VDD 850 MEM 2000.

I'll keep updating while I can, I like to sacrifice hashrate for stability on the mining rig.
